tests/fallback_test.py

```diff
@@ -306,28 +317,51 @@ def test_fallback_mid_update(self):
                     upstream_host=FallbackTest.dockerInternalIp,
                 )
             )
-            self.assert_ads_connections(
+            self.check_ads_connections_statuses(
                 client,
                 primary_status=channelz_pb2.ChannelConnectivityState.TRANSIENT_FAILURE,
                 fallback_status=channelz_pb2.ChannelConnectivityState.READY,
             )
-            stats = client.get_stats(10)
-            self.assertEqual(stats.num_failures, 0)
-            self.assertIn("server2", stats.rpcs_by_peer)
+            retryer = retryers.constant_retryer(
+                wait_fixed=datetime.timedelta(seconds=1),
+                timeout=datetime.timedelta(seconds=20),
+                check_result=lambda stats: stats.num_failures == 0
+                and "server2" in stats.rpcs_by_peer,
+            )
+            retryer(client.get_stats, 10)
             # Check that post-recovery uses a new config
             with self.start_control_plane(
                 name="primary_xds_config_run_2",
                 port=self.bootstrap.primary_port,
                 upstream_port=server3.port,
             ):
-                self.assert_ads_connections(
+                self.check_ads_connections_statuses(
                     client,
                     primary_status=channelz_pb2.ChannelConnectivityState.READY,
                     fallback_status=None,
                 )
-                stats = client.get_stats(20)
-                self.assertEqual(stats.num_failures, 0)
-                self.assertIn("server3", stats.rpcs_by_peer)
+                retryer = retryers.constant_retryer(
+                    wait_fixed=datetime.timedelta(seconds=1),
+                    timeout=datetime.timedelta(seconds=20),
+                    check_result=lambda stats: stats.num_failures == 0
+                    and "server3" in stats.rpcs_by_peer,
+                )
+                retryer(client.get_stats, 10)
+
+    def check_ads_connections_statuses(
+        self, client, primary_status, fallback_status
+    ):
+        retryer = retryers.constant_retryer(
+            wait_fixed=datetime.timedelta(seconds=1),
+            timeout=datetime.timedelta(seconds=60),
```

Comment:
Please note that expect_channel_status already does some polling: https://github.com/grpc/psm-interop/blob/5465f4b7d675862fd253fa54f7ebc53d2e64d6c8/framework/helpers/docker.py#L303-L312

It polls every `0.3` seconds  (`_STATUS_POLL_INTERVAL_MS`) with total timeout `15` seconds (_STATUS_TIMEOUT_MS). And we have two calls to this method in the check function.

Meaning with this setup in case both call fail, we're not even going to have a chance to do the second retry. Let's bump it up a bit, something like


```suggestion
            timeout=datetime.timedelta(minutes=3),
```

Reply:
Done.